publisher-provider/publisher-provider-app/src/main/java/nl/idgis/publisher/provider/sde/SDEUtils.java

```diff
@@ -7,60 +7,121 @@
 import java.util.Optional;
 import java.util.stream.Stream;
 
-import nl.idgis.publisher.provider.database.messages.ColumnFilter;
-import nl.idgis.publisher.provider.database.messages.CompoundFilter;
-import nl.idgis.publisher.provider.database.messages.DatabaseColumnInfo;
-import nl.idgis.publisher.provider.database.messages.FetchTable;
-import nl.idgis.publisher.provider.database.messages.Filter;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+
+import nl.idgis.publisher.provider.database.DatabaseType;
+import nl.idgis.publisher.provider.database.messages.*;
 import nl.idgis.publisher.provider.protocol.Record;
 import nl.idgis.publisher.provider.protocol.Records;
 
 final class SDEUtils {
-	
-	private SDEUtils() {}
-	
-	static Filter getItemsFilter() {
+
+	private final DatabaseType databaseVendor;
+
+	private final String metadataTable;
+
+	private final String columnType;
+
+	private final String columnUUID;
+
+	private final String columnPhysicalName;
+
+	private final String columnDocumentation;
+
+	private final String columnTypeDataType;
+
+	private final String columnUUIDDataType;
+
+	private final String columnPhysicalNameDataType;
+
+	private final String columnDocumentationDataType;
+
+	SDEUtils(Config databaseConfig) throws ConfigException {
+
+
+
+		if (databaseConfig.hasPath("vendor")) {
+			try {
+				this.databaseVendor = DatabaseType.valueOf(databaseConfig.getString("vendor").toUpperCase());
+			} catch(IllegalArgumentException iae) {
+				throw new ConfigException.BadValue("vendor", "Invalid vendor supplied in config");
+			}
+		} else {
+			this.databaseVendor = DatabaseType.ORACLE;
+		}
+
+		if (databaseVendor == DatabaseType.POSTGRES) {

```

Comment:
Ook hier geldt dat deze info wellicht in de enum te plaatsen is.

Reply:
Zie hierboven